### DISCUSSION TITLE
Changes in root_path behavior makes 404

### DISCUSSION THREAD

**Discussion by bayanist01** [fastapi/fastapi#11735] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI(
    debug=config.development,
    root_path='/',
)

# this opens all the endpoints in http://localhost:8000//docs  instead of http://localhost:8000/docs, i got 404 to all my requests
```

### Description

https://fastapi.tiangolo.com/advanced/behind-a-proxy/#about-root_path  

the docs says it shouldn't impact my app, but it did 

root_path variable equals to '' works perfectly, but '/' isn't working, or works as http://localhost:8000//path/to/my/endpoint

### Operating System

Linux, macOS

### FastAPI Version

0.111.0

### Pydantic Version

1.10.16

### Python Version

3.12

**Comment by YuriiMotov** [comment, marked as answer]
That was the issue of Starlette.
To resolve it upgrade FastAPI to `>=0.112.1` and then upgrade Starlette
